### PR TITLE
Added async versions of IRepository methods

### PR DESCRIPTION
**Pull request by wintondeshong**
Additionally broke out various CRUD aspects into smaller repository interfaces

Fixes issue #73 

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [n/a] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

**Comment by wintondeshong** [comment]
Don't merge until https://github.com/AndcultureCode/AndcultureCode.CSharp.Data.SqlServer/issues/5 at a minimum has a draft PR about ready to go. Don't want to keep folks from updating without providing an implementation.

File: src/AndcultureCode.CSharp.Core/Interfaces/Data/IRepository.cs [hunk 1]
```diff
@@ -1,289 +1,24 @@
...
 namespace AndcultureCode.CSharp.Core.Interfaces.Data
 {
     /// <summary>
-    ///
+    /// Handles CRUD of <typeparamref name="T"/> data
     /// </summary>
-    /// <typeparam name="T"></typeparam>
-    public interface IRepository<T>
-        where T : class, IEntity
+    public interface IRepository<T> :
```
Comment (brandongregoryscott) [inline src/AndcultureCode.CSharp.Core/Interfaces/Data/IRepository.cs:8]: Awesome refactor to split out the repository interfaces :clap:

**Review by brandongregoryscott — APPROVED** [review]

**Review by myty — APPROVED** [review]

File: src/AndcultureCode.CSharp.Core/Interfaces/Data/IRepository.cs (hunk 1 above)
Comment (wintondeshong) [reply src/AndcultureCode.CSharp.Core/Interfaces/Data/IRepository.cs:8]: Thanks @brandongregoryscott - yeah, adding the async variants further displayed the need to break these out - it was over 500 some odd lines before splitting it.

**Comment by wintondeshong** [comment]
I'm merging this into a future v1.0.0beta branch so I can make life easier while staying out of the way of others progress on new features. Once dependent issues are completed it will be re-aligned.

**Pull request merged**